tiup: add documents of mirror-set and uninstall and update command (#5038)
pingcap/docs: tiup/tiup-command-uninstall.md

---
title: tiup uninstall
---

# tiup uninstall

The `tiup uninstall` command is used to uninstall the installed components.

## Syntax

```sh
tiup uninstall <component1>:<version> [component2...N] [flags]
```

> - `<component1>`: the name of a component to uninstall.

Reviewer: When we use this command, we are very clear which component we are gonna uninstall.
```suggestion
- `<component1>`: the name of the component to uninstall.
```

> - `<version>`: the version to uninstall. If omitted, it uninstalls all installed versions of a component. For security reasons, the `--all` option must be added when `<version>` is omitted, indicating that all versions of a component need to be uninstalled.

Reviewer: ```suggestion
- `<version>`: the version to uninstall. If this field is omitted, all installed versions of the component are uninstalled. For security reasons, the `--all` option must be added when `<version>` is omitted, indicating that all versions of a component need to be uninstalled.
```

- `[component2...N]`: multiple components or versions to uninstall.

## Options

### --all

> - Uninstalls all installed versions of the specified components. You can use this option when `<version>` is omitted.

Reviewer: ```suggestion
- Uninstalls all installed versions of the specified component(s). You must use this option when `<version>` is omitted.
```

- Data type: `BOOLEAN`
- Default: `false`

### --self

> - Uninstalls TiUP itself: delete all data downloaded from the mirror, but keep the data generated by TiUP and its components. The data is stored in the directory specified by the `TIUP_HOME` environment variable. If `TIUP_HOME` is not set, the default value is `~/.tiup/`.

Reviewer: @lucklove Please help clarify what is "the data stored in the directory". Is it the data to be deleted or retained? Thanks~
```suggestion
- Uninstalls TiUP itself. When this option is used, all data downloaded from the mirror is deleted, but the data generated by TiUP and its components is retained. The data is stored in the directory specified by the `TIUP_HOME` environment variable. If `TIUP_HOME` is not set, the default value is `~/.tiup/`.
```

- Data type: `BOOLEAN`
- Default: `false`

## Outputs

> - If exits normally, output `Uninstalled component "%s" successfully!`.

Reviewer: ```suggestion
- If the command exits without any error, `Uninstalled component "%s" successfully!` is output.
```
